### PR TITLE
Update to lens_plot.py

### PR DESCRIPTION
**Pull request by jocelyn-mcmahon-16**
Added feature to lens_plot.py to allow for the plotting of multiple sources/images in the point_source_plot function. Allows for input of parameters _name_list_ and _index_, which respectively specify a new list of labels to be used for the images and, when used with multiple sources, labels each source's images with a number to more easily determine which image comes from which source.

### CODE REVIEW

File: lenstronomy/Plots/lens_plot.py [hunk 1]
```diff
@@ -37,7 +39,9 @@ def lens_model_plot(
     coord_center_dec=0,
     coord_inverse=False,
     fast_caustic=True,
-    **kwargs
+    name_list=None,
```
Comment (sibirrer) [inline lenstronomy/Plots/lens_plot.py:42]: can you document this new input variables in the doc strings

File: lenstronomy/Plots/lens_plot.py [hunk 2]
```diff
@@ -283,12 +290,36 @@ def point_source_plot(
...
+    :param index: number of sources, an integer number. Default None.
     :param kwargs: additional plotting keyword arguments
     :return: matplotlib axis instance with figure
     """
     from lenstronomy.LensModel.Solver.lens_equation_solver import LensEquationSolver
 
+    if name_list is None and index is None:
+        name_list = _NAME_LIST
+    elif name_list is None and index is not None:
+        name_list = [
```
Comment (sibirrer) [inline lenstronomy/Plots/lens_plot.py:303]: this line of code is not tested. Can you add a test that goes through this option?

Comment (sibirrer) [reply lenstronomy/Plots/lens_plot.py:303]: do you really need this line of code? can't you just get name_list = _NAME_LIST and then also go through the for-loop the same way as for custom input for name_list?

File: lenstronomy/Plots/lens_plot.py [hunk 3]
```diff
@@ -283,12 +290,36 @@ def point_source_plot(
...
+            f"{index+1}E",
+            f"{index+1}F",
+            f"{index+1}G",
+            f"{index+1}H",
+            f"{index+1}I",
+            f"{index+1}J",
+            f"{index+1}K",
+        ]
+    elif name_list is not None and index is None:
+        name_list = name_list
```
Comment (sibirrer) [inline lenstronomy/Plots/lens_plot.py:317]: can you also make a test case with this option?

**Review by sibirrer — CHANGES_REQUESTED** [review]
Looks great, thank you @jocelyn-mcmahon-16 ! I have some suggestions and a request for better test coverage

File: lenstronomy/Plots/lens_plot.py (hunk 3 above)
Comment (jocelyn-mcmahon-16) [reply lenstronomy/Plots/lens_plot.py:309]: Added a test for this one!

File: lenstronomy/Plots/lens_plot.py (hunk 2 above)
Comment (jocelyn-mcmahon-16) [reply lenstronomy/Plots/lens_plot.py]: I added a test for each case. I also updated it to loop over _NAME_LIST like you suggested, but I left it as a separate case.

File: lenstronomy/Plots/lens_plot.py (hunk 1 above)
Comment (jocelyn-mcmahon-16) [reply lenstronomy/Plots/lens_plot.py:42]: Added the variables!

**Review by sibirrer — APPROVED** [review]
Thank you very much @jocelyn-mcmahon-16 !

**Pull request merged**